### DISCUSSION TITLE
Question about the Vin

### DISCUSSION THREAD

**Discussion by cymqqqq** [RGB-WG/rgb#102] [Q&A] [answered]
Hi there, I found the VIn code in the rgb-core/src/contract/bundile.rs. v0.11: line 35 pub type Vin = Vout;
What's the definition of the Vin? Or how we can use it in our project? Any docs?
And I found BundleItem is deleted instead of TransitionBundle, which is composed of 
pub struct TransitionBundle {
    pub input_map: Confined<BTreeMap<Vin, OpId>, 1, U16>,
    pub known_transitions: Confined<BTreeMap<OpId, Transition>, 1, U16>,
}
So what's the definition of the input_map?

**Comment by cymqqqq** [comment]
Like this?
"Vin" : [
 {
"txid": "8adbca5e652c68f8f3c30ac658115bc4af395d0cc7e6beaea18168295c29d011",
"vout": 0,
"scriptSig": "<our scriptSig>"
 }
]

**Comment by dr-orlovsky** [comment, marked as answer]
Vin means the ordinal number of some transaction input. It is just a u32 integer. It is defined that way since it is the same type int as Vout.

Your comment
> Like this?
> "Vin" : [
> {
> "txid": "8adbca5e652c68f8f3c30ac658115bc4af395d0cc7e6beaea18168295c29d011",
> "vout": 0,
> "scriptSig": ""
> }
> ]

does not make any sense to me

> So what's the definition of the input_map?

The definition of the input map is exactly given in the code:
```rust
  pub input_map: Confined<BTreeMap<Vin, OpId>, 1, U16>,
```
